### PR TITLE
daemon,overlord/snapstate: back /snaps (and so snap list) using state

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

**Comment by pedronis** [comment]
retest this please

File: snappy/overlord.go [hunk 1]
```diff
@@ -413,7 +406,12 @@ func (o *Overlord) InstallWithSideInfo(snapFilePath string, sideInfo *snap.SideI
 		return nil, err
 	}
 
-	// we need this for later
+	// XXX: do this here as long as we are somewhat manifest based
```
Comment (pedronis) [inline snappy/overlord.go]: I just mvoed this but this comment needs rephasing now I think

**Comment by chipaca** [comment]
:+1:

**Comment by niemeyer** [comment]
LGTM

**Pull request merged**